src/base1/test-dbus-common.js

```diff
@@ -1130,4 +1130,28 @@ function dbus_track_tests(channel_options, bus_name) {
                     });
             });
     });
+
+    QUnit.asyncTest("receive fd", function() {
+        assert.expect(4);
+
+        var dbus = cockpit.dbus("com.redhat.Cockpit.DBusTests.Test", channel_options);
+        dbus.call("/otree/frobber", "com.redhat.Cockpit.DBusTests.Frobber", "MakeTestFd", []).
+            done(function (reply) {
+                var fd = reply[0];
+                assert.equal(typeof(fd.internal), 'string');
+                assert.equal(fd.payload, 'stream');
+
+                var channel = cockpit.channel(fd);
+                QUnit.stop();
+                channel.onmessage = function (event, data) {
+                    assert.equal(data, 'Hello, fd');
+                    channel.close();
+                    QUnit.start();
```

Comment:
Yes, start/stop are refcounted. We need this here because the `done()` path kicks off another async call to wait for the first message. We still need it in `always()` in case the dbus call fails. I'll add a comment.